storage/src/macros.rs

```diff
@@ -0,0 +1,29 @@
+
+/// Macro to register and use a metric with description and labels.
+/// This macro is a wrapper around the `metrics` crate's `counter!` and `describe_counter!`
+/// macros. It ensures that the description is registered only once.
+///
+/// Usage:
+///   firewood_metric!("metric_name", "description", value_to_increment_by)
+///   firewood_metric!("metric_name", "description", value_to_increment_by, "label" => "value")
+#[macro_export]
+macro_rules! firewood_metric {
+    // With labels
+    ($name:expr, $desc:expr, $inc:expr, $($labels:tt)+) => {
+        {
+            static METRIC: std::sync::LazyLock<()> = std::sync::LazyLock::new(|| {
```

Comment:
I like the use of `LazyLock` to get these added only once at runtime.

Let's switch to using `std::sync::Once` since you're not actually initializing a variable. [`call_once`](https://doc.rust-lang.org/stable/std/sync/struct.Once.html#method.call_once) does exactly the same thing (`LazyLock` uses `Once` [itself](https://doc.rust-lang.org/stable/src/std/sync/lazy_lock.rs.html#67)) and we can use it directly here since we don't need to retain the ZST or even a pointer to it.